code/modules/surgery/organs/augments_arms.dm

```diff
@@ -411,3 +411,128 @@
+	disabled = TRUE
+	force = force_when_disabled
+	addtimer(CALLBACK(src, PROC_REF(reboot)), 10 SECONDS)
+
+/obj/item/shield/v1_arm/proc/reboot()
+	disabled = FALSE
+	force = initial(force)
+
+/obj/item/shield/v1_arm/add_parry_component()
+	AddComponent(/datum/component/parry, _stamina_constant = 2, _stamina_coefficient = 0.35, _parryable_attack_types = ALL_ATTACK_TYPES, _parry_cooldown = (1 / 3) SECONDS, _no_parry_sound = TRUE) // 0.3333 seconds of cooldown for 75% uptime, countered by ions and plasma pistols
+
+/obj/item/shield/v1_arm/hit_reaction(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", final_block_chance = 0, damage = 0, attack_type = MELEE_ATTACK)
+	if(disabled)
+		return FALSE
+	// Hit by a melee weapon or blocked a projectile
+	. = ..()
+	if(.) // they did parry the attack
+		// Hit by a projectile
+		if(. > 1) // a perfect parry
+			set_light(3)
+			addtimer(CALLBACK(src, TYPE_PROC_REF(/atom, set_light), 0), 0.25 SECONDS)
+
+			if(istype(hitby, /obj/item/projectile))
+				var/obj/item/projectile/P = hitby
+				if(P.shield_buster || istype(P, /obj/item/projectile/ion)) //EMP's and unpariable attacks, after all.
+					return FALSE
+				if(P.reflectability == REFLECTABILITY_NEVER) //only 1 magic spell does this, but hey, needed
+					owner.visible_message("<span class='danger'>[owner] blocks [attack_text] with [src]!</span>")
+					playsound(src, 'sound/weapons/effects/ric3.ogg', 100, TRUE)
+					return TRUE
+
+				P.damage = max(min(reflect_damage_cap, P.damage += 10), P.damage)
+				var/sound = pick('sound/effects/explosion1.ogg', 'sound/effects/explosion2.ogg', 'sound/effects/meteorimpact.ogg')
+				P.hitsound = sound
+				P.hitsound_wall = sound
+				P.add_overlay("parry")
+				playsound(src, 'sound/weapons/v1_parry.ogg', 100, TRUE)
+				owner.visible_message("<span class='danger'>[owner] parries [attack_text] with [src]!</span>")
+				add_attack_logs(P.firer, src, "hit by [P.type] but got parried by [src]")
+				return -1
+
+			owner.visible_message("<span class='danger'>[owner] parries [attack_text] with [src]!</span>")
+			playsound(src, 'sound/weapons/v1_parry.ogg', 100, TRUE)
+			if(istype(hitby, /obj/item)) //Thrown items
+				var/obj/item/TT = hitby
+				addtimer(CALLBACK(TT, TYPE_PROC_REF(/atom/movable, throw_at), locateUID(TT.thrownby), 15, 15, owner), 0.1 SECONDS)//yeet that shit right back
+				return TRUE
+			melee_attack_chain(owner, hitby)
+			return TRUE
+		else
+			owner.visible_message("<span class='danger'>[owner] blocks [attack_text] with [src]!</span>")
+			playsound(src, 'sound/weapons/effects/ric3.ogg', 100, TRUE)
+			return TRUE
+
+/obj/item/v1_arm_shell
+	name = "vortex feedback arm implant frame"
+	desc = "An implant awaiting installation of a vortex anomaly core"
+	icon_state = "v1_arm"
+
+/obj/item/v1_arm_shell/attackby(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/assembly/signaler/anomaly/vortex))
+		var/obj/item/assembly/signaler/anomaly/vortex/V = I
```

Comment:
you don't need to cast this to a var since all you do is use it to get its name in the `to_chat()` you can just do `[I]` and will get the same output.

Reply:
Done